pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -414,6 +424,72 @@ func parseShootedSeedBackup(settings map[string]string) (*gardencorev1beta1.Seed
 	return backup, nil
 }
 
+func parseShootedSeedProviderConfig(settings map[string]string) (*runtime.RawExtension, error) {
```

Comment:
Can you reuse `unstructured.SetNestedField` and `json.Marshal` instead of reimplementing this stuff on our side?

I think this func should look something like this:
```go
func parseShootedSeedProviderConfig(settings map[string]string) (*runtime.RawExtension, error) {
	// reconstruct providerConfig structure
	providerConfig := map[string]interface{}{}

	for k, v := range settings {
		if strings.HasPrefix(k, "providerConfig.") {
			if err := unstructured.SetNestedField(providerConfig, v, strings.Split(k, ".")[1:]...); err != nil {
				return nil, err
			}
		}
	}

	jsonStr, err := json.Marshal(providerConfig)
	if err != nil {
		return nil, err
	}

	return &runtime.RawExtension{
		Raw: jsonStr,
	}, nil
}
```